joomla/joomla-cms
# [4.0] Specify title and icon of dashboards in the component manifest

This is an alternative solution to https://github.com/joomla/joomla-cms/pull/31392 which also takes care of the dashboard icons.
Inspired by comment https://github.com/joomla/joomla-cms/pull/31392#issuecomment-726646464

### Summary of Changes
Instead of guessing a language string for the title and the icon (!) of any dashboard, this PR changes it to lookup up the components manifest file and check the `<dashboard>` entries in there. If a corresponding entry for the current requested dashboard is found, then the title and icon attributes are taken as title and icon.

If not entry is found, the current behavior with guessed language strings for titles and icons are used. I'm not sure if we want to keep the icon stuff as it looks very fishy to me. The title guessing may be fine to keep.


### Testing Instructions
Open the various dashboards (users, content, privacy, system, help, home, components, menus) and make sure the icons and title are the same as bevore


### Actual result BEFORE applying this Pull Request
Title and icon are taken from guessed language strings or hardcoded in the view.


### Expected result AFTER applying this Pull Request
Title and icon are taken from the manifest files where they exist (home, system, help, components have no corresponding component)


### Documentation Changes Required
dashboard developer documentation needs to reflect this.
